src/main/java/jek/collections/IntArrayList.java

```diff
@@ -0,0 +1,155 @@
+package jek.collections;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+public class IntArrayList implements List {
+
+    private int end;
+    private Object[] ial;
+
+    public IntArrayList() {
+        ial = new Object[10];
+        end = 0;
+    }
+
+    @Override
+    public int size() {
+        return end;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        if (end == 0)
+            return true;
+        return false;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        for (Object search : ial)
+            if (search.equals(o))
+                return true;
+
+        return false;
+    }
+
+    @Override
+    public Iterator iterator() {
+        return null;
+    }
+
+    @Override
+    public Object[] toArray() {
+        return new Object[0];
+    }
+
+    @Override
+    public boolean add(Object o) {
+        if(end == ial.length) {                             //чекаем, если конец равен длинне
+            Object[] ial2 = new Object[ial.length*2];       //создаем новый массив и копируем
+            for (int i = 0; i < ial.length; i++)
```

Comment:
Let's also start checking out what is there for us in the related libraries. Please check methods like System.arrayCopy
or Arrays.copy  - those methods are usually written in super efficient manner and so will make your code shorter(less lines) and faster to execute.